lgc/patch/PatchPeepholeOpt.cpp

```diff
@@ -143,4 +148,35 @@ void PatchPeepholeOpt::visitIntToPtr(IntToPtrInst &intToPtr) {
     m_instsToErase.push_back(binaryOperator);
 }
 
+// =====================================================================================================================
+// Visit a call instruction.
+//
+// Peephole log2(const +/- x) -> log2(max(0.0, const +/- x)).
+// This addresses a potential precision underflow in applications intolerant to in-spec math reordering.
+//
+// @param callInst: The call instruction to visit.
+void PatchPeepholeOpt::visitCallInst(CallInst &callInst) {
+  if (callInst.getIntrinsicID() != Intrinsic::log2)
+    return;
+
+  Value *V = callInst.getOperand(0);
+
+  if (!(match(V, m_FSub(m_Constant(), m_Value())) || match(V, m_FSub(m_Value(), m_Constant())) ||
```

Comment:
FYI there is m_c_BinOp(opcode, lhs, rhs) for matching with operands in either order.